jerry-core/vm/vm.c

```diff
@@ -211,27 +194,46 @@ vm_op_set_value (ecma_value_t object, /**< base object */
     property_p = ecma_get_prop_name_from_value (property);
   }
 
-  ecma_value_t completion_value = ECMA_VALUE_EMPTY;
-
-  if (!ecma_is_lexical_environment (object_p))
+  if (JERRY_UNLIKELY (!ecma_is_value_object (base)))
   {
-    completion_value = ecma_op_object_put (object_p,
-                                           property_p,
-                                           value,
-                                           is_strict);
+    ecma_value_t object = ecma_op_to_object (base);
+    JERRY_ASSERT (!ECMA_IS_VALUE_ERROR (object));
+
+    object_p = ecma_get_object_from_value (object);
+    ecma_op_ordinary_object_prevent_extensions (object_p);
```

Comment:
Is this standard?

Reply:
This has been added as part of #2917, it's seems to me that it's correct.